### ISSUE TITLE
Poiyomi 8.2 Optimization Errors

### ISSUE THREAD

**Issue by kklouzal** [d4rkc0d3r/d4rkAvatarOptimizer#28]
```
Error optimizing shader .poiyomi/Poiyomi 8.2/Poiyomi Pro: Index was out of range. Must be non-negative and less than the size of the collection.
Parameter name: index
  at System.ThrowHelper.ThrowArgumentOutOfRangeException (System.ExceptionArgument argument, System.ExceptionResource resource) [0x00029] in <eae584ce26bc40229c1b1aa476bfa589>:0 
  at System.ThrowHelper.ThrowArgumentOutOfRangeException () [0x00000] in <eae584ce26bc40229c1b1aa476bfa589>:0 
  at System.Collections.Generic.List`1[T].get_Item (System.Int32 index) [0x00009] in <eae584ce26bc40229c1b1aa476bfa589>:0 
  at d4rkpl4y3r.ShaderOptimizer.Run () [0x00137] in C:\Users\sysmin\AppData\Local\VRChatProjects\Maya-2\Assets\d4rkAvatarOptimizer\src\ShaderAnalyzer.cs:1980 
  at d4rkpl4y3r.ShaderOptimizer.Run (d4rkpl4y3r.ParsedShader source, System.Collections.Generic.Dictionary`2[TKey,TValue] staticPropertyValues, System.Int32 mergedMeshCount, System.Collections.Generic.List`1[T] mergedMeshNames, System.Collections.Generic.HashSet`1[T] defaultAnimatedProperties, System.Collections.Generic.List`1[T] mergedMeshIndices, System.Collections.Generic.Dictionary`2[TKey,TValue] arrayPropertyValues, System.Collections.Generic.Dictionary`2[TKey,TValue] texturesToNullCheck, System.Collections.Generic.HashSet`1[T] texturesToMerge, System.Collections.Generic.Dictionary`2[TKey,TValue] animatedPropertyValues, System.Collections.Generic.List`1[T] setKeywords) [0x001c5] in C:\Users\sysmin\AppData\Local\VRChatProjects\Maya-2\Assets\d4rkAvatarOptimizer\src\ShaderAnalyzer.cs:1025 
UnityEngine.Debug:LogError (object)
d4rkpl4y3r.ShaderOptimizer:Run (d4rkpl4y3r.ParsedShader,System.Collections.Generic.Dictionary`2<string, string>,int,System.Collections.Generic.List`1<string>,System.Collections.Generic.HashSet`1<System.ValueTuple`2<string, bool>>,System.Collections.Generic.List`1<int>,System.Collections.Generic.Dictionary`2<string, System.ValueTuple`2<string, System.Collections.Generic.List`1<string>>>,System.Collections.Generic.Dictionary`2<string, string>,System.Collections.Generic.HashSet`1<string>,System.Collections.Generic.Dictionary`2<string, string>,System.Collections.Generic.List`1<string>) (at Assets/d4rkAvatarOptimizer/src/ShaderAnalyzer.cs:1029)
d4rkAvatarOptimizerEditor/<>c__DisplayClass62_0:<CreateOptimizedMaterials>b__1 (int) (at Assets/d4rkAvatarOptimizer/src/d4rkAvatarOptimizerEditor.cs:1253)
System.Threading._ThreadPoolWaitCallback:PerformWaitCallback ()
```

Many of my meshes fail to combine when using Poiyomi 8.2
I had started adding meshes that failed to the exclusions, but it was failing on almost every single one (some passed fine).
[err.unitypackage.gz](https://github.com/d4rkc0d3r/d4rkAvatarOptimizer/files/10842106/err.unitypackage.gz)

**Comment by kklouzal** [comment]
I deleted the optimized copy of my avatar from the hierarchy, and removed d4rkAvatarOptimizer from my project, relaunched Unity, and went to build my avatar. It's been over 2 hours now compiling shader variants and it looks like its got a long long way to go still.

I'm not sure what happened here but something isn't right.

**Comment by kklouzal** [comment]
Is there a way to undo the
https://github.com/d4rkc0d3r/d4rkAvatarOptimizer#write-properties-as-static-values
?

**Comment by d4rkc0d3r** [comment]
The optimizer doesn't touch the original assets (aside from the button that converts normal maps to BC5).

The write properties as static values only applies to the shaders that get generated by the optimizer that end up in the TrashBin folder.

**Comment by kklouzal** [comment]
humm okay, i looked through the .mat files and it looks like it changed it to have static values. I figured that was what caused it to slow down shader compilation.
After letting it run completely (4 hours) each subsequent run here after has been relatively quick, 5-10 minutes.

Still though, the poiyomi 8.2 shaders dont seem to play nice? Ended up having to exclude almost every mesh that used them from optimization..
